### PR TITLE
setting up a pre-commit hook

### LINKED ISSUE #22: Set up test runner, precommit, and linter
- [x] test runner
- [x] linter
- [x]  pre-commit 
- [x] istanbul

### PR DESCRIPTION
**Pull request by samhstn**
Fixes #22

### CODE REVIEW

**Comment by jrans** [comment]
@shouston3 Not sure this is a PR right now as you've already pushed this to master but please add to PR immediately with: 
- [x] eslint in dev-dependencies
- [x] istanbul.yml file
- [x] change test script to either 
  -  run test files with duffix `.test.js` not just `.js` as stops us from adding helpers or fixtures to the test folder
  - The proposed solution in #23 where we use index.js as the runner and test command runs just that file which will require in the rest.

**Comment by jrans** [comment]
an example of a  `.istanbul.yml` file is 

```
verbose: false
instrumentation:
  root: ./src
  excludes: []
  include-all-sources: true
check:
  global:
    branches   : 100
    functions  : 100
    lines      : 100
    statements : 100
```

which reduces the package.json command also

**Comment by jrans** [comment]
@shouston3 conflicts to be resolved

**Comment by jrans** [comment]
@shouston can you mention #22 in your commit messages

**Comment by samhstn** [comment]
Fixes #22

File: package.json [hunk 1]
```diff
@@ -15,14 +15,15 @@
...
     "pre-commit": "^1.1.3",
     "tape": "^4.6.0"
   },
   "scripts": {
-    "test": "tape ./test/*.js",
-    "coverage": "istanbul cover tape ./test/*.js && istanbul check-coverage --statements 100 --functions 100 --lines 100 --branches 100",
-    "eslint": "eslint ."
+    "test": "node_modules/.bin/tape ./src/test/*.js",
+    "eslint": "node_modules/.bin/eslint .",
+    "coverage": "node_modules/.bin/istanbul cover node_modules/.bin/tape ./src/test/*.js"
```
Comment (jrans) [inline package.json]: you've duplicated the test command, how about "node_modules/.bin/istanbul cover npm run test"

**Comment by jrans** [comment]
may want a .eslintignore file

File: package.json (hunk 1 above)
Comment (samhstn) [reply package.json]: 👍

**Comment by samhstn** [comment]
Added the `.eslintignore` file ⬆️

File: package.json [hunk 2]
```diff
@@ -15,17 +15,19 @@
...
+    "eslint": "^3.5.0",
     "istanbul": "^0.4.5",
     "pre-commit": "^1.1.3",
     "tape": "^4.6.0"
   },
   "scripts": {
-    "test": "tape ./test/*.js",
-    "coverage": "istanbul cover tape ./test/*.js && istanbul check-coverage --statements 100 --functions 100 --lines 100 --branches 100",
-    "eslint": "eslint ."
+    "test": "node_modules/.bin/tape ./src/test/*.test.js",
```
Comment (jrans) [inline package.json]: `test/*.test.jsa and change the location of test folder

**Review by eliasmalik — APPROVED** [review]

**Comment by jrans** [comment]
👍 Well done Sam

**Comment by nelsonic** [comment]
@jrans not a _fan_ of having _another_ configuration file in the project ...
what does `.istanbul.yml` give us that cannot be included in the _script_ in `package.json` ...
e.g: https://github.com/dwyl/hapi-auth-jwt2/blob/c2fe39237e1f1c6dc2ae2d6aae38e00dc343ded7/package.json#L61

**Comment by samhstn** [comment]
reverted back to having all the istanbul config in coverage script in #29

**Pull request merged**